projects/forms/src/lib/abstracts/save-on-exit/save-on-exit.component.abstract.ts

```diff
@@ -0,0 +1,50 @@
+import { Directive, HostListener, OnDestroy } from '@angular/core';
+import { Subject } from 'rxjs';
+
+@Directive()
+export abstract class NgxSaveOnExitComponent implements OnDestroy {
```

Comment:
Zouden alle public methods niet `protected` moeten zijn? Deze component gaat altijd geïmplementeerd worden, dus de public methods zijn sowieso protected in a way.

Reply:
Ik heb ze zowel hier als bij de service bewust niet protected gemaakt omdat ik eigenlijk niet wil bepalen dat die methodes niet erbuiten kunnen gebruikt worden.

De isValid of de isDirty methode bv kunnen perfect bruikbaar zijn voor andere components via een ViewChild bv, ik vind het een beetje te limiterend om die dan protected te gaan maken en users te gaan forceren om dan zo van die "isCompontentDirty" methodes te laten schrijven die dan op hun beurt gewoon het resultaat van de "isDirty" teruggeeft.

Zelfde geldt voor de SaveOnExitService, ik wil daar mensen niet gaan inboxen op voorhand. Was dit voor een intern project waar we dat altijd in de hand hebben en ten alle tijde zelf kunnen beslissen om het aan te passen, dan ging ik het ook protected gemaakt hebben. Maar gezien het hier om een public package gaat en deze methodes allemaal wel een functie kunnen hebben buiten de flow hier, laat ik ze liever publiek.